### DISCUSSION TITLE
How to use a graphql codegen mutation in flutter?

### DISCUSSION THREAD

**Discussion by LiamKarlMitchell** [heftapp/graphql_codegen#251] [Q&A]
How can we execute a generated mutation with some input variables in Flutter?
I could only see examples for querying in the docs and example folder.

Could someone please provide some examples for widget, hook & direct from the client?

Would like to use the results in Builder if possible but could not find a way to see when it was completed other than the onComplete method.
Is there a way to get a hasResults or isLoadedWithResults in the builder functions? As sometimes the data might come from cache or eagerly set by dev?

How about handling onError as well, to inform the user something went wrong?

And how to configure the cache for queries, update it for paginated use case's?
Is wiping the cache as easy as just clearing hive?

For example given the schema.

Client side gql file.
```gql
mutation SetSomeValue($input: SetSomeValueInput!) {
  setSomeValue(input: $input)
}
```

Server side (Lighthouse Laravel) schema.
```
enum SetSomeValueStatus {
    """TOKEN_SET"""
    TOKEN_SET

    """TOKEN_ERASED"""
    TOKEN_ERASED

    """TOKEN_NOT_SET"""
    TOKEN_NOT_SET
}

# noinspection GraphQLUnresolvedReference,GraphQLMissingType
input SetSomeValueInput @trim {
    token: String! @rules(apply: ["max:255"])
}

# noinspection GraphQLUnresolvedReference,GraphQLMissingType
extend type Mutation {
    "Set some value for logged in user."
    SetSomeValue(input: SetSomeValueInput @spread): SetSomeValueStatus! @guard
    @field(resolver: "SetSomeValue")
}
```

It gets generated into a bunch of code probably too big to paste here.

```dart
Variables$Mutation$SetSomeValue
Mutation$SetSomeValue
documentNodeMutationSetSomeValue
Options$Mutation$SetSomeValue
WatchOptions$Mutation$SetSomeValue
Mutation$SetSomeValue$HookResult
WidgetOptions$Mutation$SetSomeValue
RunMutation$Mutation$SetSomeValue
Builder$Mutation$SetSomeValue
Mutation$SetSomeValue$Widget
```

I had Mutation kind of working.
But it was getting called constantly for some reason.

**Comment by budde377** [comment]
Try and take a look at how `graphql_flutter` is working. You can then use the code generator to generate a builder.
